internal/core/service/flight_service.go

```diff
@@ -0,0 +1,120 @@
+package service
+
+import (
+	"sort"
+
+	"github.com/cyneptic/letsgo/infrastructure/provider"
+	"github.com/cyneptic/letsgo/internal/core/entities"
+	"github.com/cyneptic/letsgo/internal/core/ports"
+)
+
+type FlightService struct {
+	pv ports.SortFilterProviderContract
+}
+
+func NewFlightService() *FlightService {
+	pv := provider.NewSortFilterProviderClient("http://localhost:8000")
+	return &FlightService{
+		pv: pv,
+	}
+}
+
+// Todo Filter
+func (f *FlightService) FilterFlightList(source string, destination string, departure string, PlaneType string, t1 int, t2 int, RemainSeat string) []entities.Flight {
+	flights, err := f.pv.RequestFlight(source, destination, departure)
+	if err != nil {
+		panic(err)
+	}
+
+	var filteredListed []entities.Flight
+	var remainFligtlist []entities.Flight
+	var hourFilterFlight []entities.Flight
+
+	//?filter
+	if RemainSeat == "true" {
```

Comment:
remainSeat is a number, client will filter based on the number of remaining seats:
I want all flights with 5 empty seats so I can get tickets for 5 seats.